qiskit/circuit/quantumcircuit.py

```diff
@@ -205,12 +205,12 @@ def data(self):
 
     @property
     def calibrations(self):
-        """Return calibration dictionary.
+        """Return calibration default dictionary.
 
         The custom pulse definition of a given gate is of the form
             {'gate_name': {(qubits, params): schedule}}
         """
-        return dict(self._calibrations)
+        return self._calibrations
```

Comment:
With this implementation, something like this:
```
cals = qc.calibrations
dne_gates = cals['DNE']
```
would return an empty dict rather than raising a `KeyError`. This isn't that bad, and trying to grab any cal from `dne_gates` would then fail, but it's strange to build a `calibrations` which contains all the keys for which you've tried to key into.

We could update the setter below to turn the input into a `defaultdict(dict)`, and then the other files would stay the same. 

I'm not as concerned about returning `self._calibrations` from the dag, since I believe it's not as often directly accessed by users